webhooks-extension/endpoints/webhook.go

```diff
@@ -126,12 +148,142 @@ func (r Resource) createWebhook(request *restful.Request, response *restful.Resp
 		return
 	}
 	webhooks[webhook.Name] = webhook
+	logging.Log.Debugf("Writing the GitHubSource webhook ConfigMap in namespace %s", installNs)
 	r.writeGitHubWebhooks(installNs, webhooks)
 	response.WriteHeader(http.StatusCreated)
 }
 
+// Removes from ConfigMap, removes the actual GitHubSource, removes the webhook
+func (r Resource) deleteWebhook(request *restful.Request, response *restful.Response) {
+	logging.Log.Debug("In deleteWebhook")
+	name := request.PathParameter("name")
+	namespace := request.QueryParameter("namespace")
+	deletePipelineRuns := request.QueryParameter("deletepipelineruns")
+
+	var toDeletePipelineRuns = false
+	var err error
+
+	if deletePipelineRuns != "" {
+		toDeletePipelineRuns, err = strconv.ParseBool(deletePipelineRuns)
+		if err != nil {
+			theError := errors.New("bad request information provided, cannot handle deletepipelineruns query (should be set to true or not provided)")
+			logging.Log.Error(theError)
+			RespondError(response, err, http.StatusInternalServerError)
+		}
+	}
+
+	if namespace == "" {
+		theError := errors.New("no namespace was provided")
+		logging.Log.Error(theError)
+		RespondError(response, theError, http.StatusBadRequest)
+	}
+
+	if name != "" {
+		foundRepoURL := r.findRepoURLFromConfigMap(name, namespace)
+		// This will remove any PipelineRuns too if specified to, hence the need for the repository URL
+		err = r.deleteGitHubWebhookByName(name, namespace, foundRepoURL, toDeletePipelineRuns)
+		if err != nil {
+			if strings.Contains(err.Error(), "could not find webhook with name") {
+				logging.Log.Errorf("webhook (name %s) not found in namespace %s", name, namespace)
+				RespondError(response, err, http.StatusNotFound)
+			} else {
+				logging.Log.Errorf("error deleting the webhook (name %s), error: %s", name, err)
+				RespondError(response, err, http.StatusInternalServerError)
+			}
+		} else {
+			logging.Log.Info("Deleted the webhook OK, deleting from ConfigMap next")
```

Comment:
%s webhook name to identify which one was deleted ..... might also want to add similar to method entry after reading values.